src/client/client.ts

```diff
@@ -50,6 +51,15 @@ export interface ClientOptions {
     /** Called to create the connection to the server. */
     createMessageTransports: () => MessageTransports | Promise<MessageTransports>
 
+    /**
+     * Called when the server requests to list files or get file contents. See
+     * https://github.com/sourcegraph/language-server-protocol/blob/8313aa98092692c17faffda5336b8951fc03266a/extension-files.md
+     */
+    files?: {
+        xfiles: (base?: string) => Promise<TextDocumentIdentifier[]>
```

Comment:
Now that these are no longer LSP extensions but core parts of CXP, you should remove the `x` prefix from these names.

Also, the docs should be ready for prime-time (ie no linking to random Markdown files on a commit, unless you add a TODO).